src/query/api/v1/handler/prometheus/native/read.go

```diff
@@ -138,11 +138,12 @@ func (h *PromReadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	result, params, respErr := h.ServeHTTPWithEngine(w, r, h.engine, &executor.EngineOptions{
-		QueryContextOptions: models.QueryContextOptions{
-			LimitMaxTimeseries: fetchOpts.Limit,
-		},
+	engineOpts := h.engine.Opts().SetQueryContextOptions(models.QueryContextOptions{
```

Comment:
why not do this at the level above this?

Reply:
I was just keeping this where it was originally which is in this function.

Reply:
Theoretically users can override the time series limit in their query, so we'd need to update the Option limit here unfortunately.